fetchRebalancerData.js

```diff
@@ -9,38 +10,52 @@ const contractAbi = require('./UniswapNFTPositionManagerAbi.json');
 
 const uniswapPositionManagerContractAddress = '0xC36442b4a4522E871399CD717aBDD847Ab11FE88';
 const contract = new web3.eth.Contract(contractAbi, uniswapPositionManagerContractAddress);
-const instaContractAddress = '0x575D9121127bEc262Bc920F465d60A9418A63744';
-let finalTransactions;
-
-const getEvents = async (fromBlock) => {
-    const transactionHashes = new Set();
-
-    await contract.getPastEvents("IncreaseLiquidity",
-        { fromBlock: 25698992, toBlock: 25774738 },
-        (err, events) => {
-            for (let i = 0; i < events.length; i++) {
-                // console.log(events[i].transactionHash);
-                transactionHashes.add(events[i].transactionHash);
-            }
-            // console.log(events[0]);
-            // transactionHash = events[0].transactionHash;
-        });
-
-    events = await contract.getPastEvents("DecreaseLiquidity",
-        { fromBlock: 25698992, toBlock: 25774738 }
-    );
-
-    // console.log(events);
-    const instaTransactionHashes = events.map(e => {
-        return {
-            transactionHash: e.transactionHash,
-            tokenId: e.returnValues.tokenId,
-            liquidity: e.returnValues.liquidity
-        };
-    }).filter(transaction => transactionHashes.has(transaction.transactionHash));
-    // console.log(instaTransactionHashes);
-
-    finalTransactions = await Promise.all(instaTransactionHashes.map(async (transaction) => {
+let transactionData;
+
+const getEvents = async (dsaAccounts) => {
+    // store all transaction hashes obtained from IncreaseLiquidity event emits in a set for quick search later on
+    const transactionHashes = new Map();
+
+    let latestBlock = await web3.eth.getBlockNumber(); // get the latest block number
+    let allValidTransactionHashes = [];
+    let currentToBlock;
+
+    for (let currentFirstBlock = 25657968; currentFirstBlock < latestBlock; currentFirstBlock = currentToBlock + 1) {
```

Comment:
why `currentFirstBlock = currentToBlock + 1`?

Reply:
`currentToBlock` is the block where the previous query ended, so the next time we need to query from one block after where the last one ended right

Reply:
But from getPastEvent we can query data from a range of blocks. 

And I think you are querying it from `currentBlock` to `currentBlock + 3000`!

So, this loop should also hop by same length.

Reply:
Actually, getPastEvents returns data from firstBlock to toBlock both inclusive. 

So, say we had currentFirstBlock as 1000, then because of the line - 
`currentToBlock = Math.min(latestBlock, currentFirstBlock + 3000);` at the start of the `for` loop block, `currentToBlock` gets set to 1000 + 3000 = 4000. Then, when we move to the next iteration, we set `currentFirstBlock` to `currentToBlock + 1` (i.e., 4001), which is correct since we have already covered blocks before 4001!

Reply:
Actually, getPastEvents returns data from `currentFirstBlock` to `currentToBlock` both inclusive, that's why we need to start the next iteration from `currentToBlock + 1`.

At the start of the block for the loop, we set `currentToBlock = Math.min(latestBlock, currentFirstBlock + 3000);`, so, say if the `currentFirstBlock` was 1000, `currentToBlock` with get set to 4000. When this block ends, `currentFirstBlock` will get set to `currentToBlock + 1`, i.e. 4001, which means we did not miss any block, and we didn't have to repeat any either!